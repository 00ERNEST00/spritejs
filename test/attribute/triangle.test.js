import TriangleAttr from '../../src/attribute/triangle';

describe('attribute set get', () => {
  const subject = {
    onPropertyChange(key, value, oldValue, attributes) { },
  };
  const attr = new TriangleAttr(subject);

  test('get set', () => {
    expect(attr.d).toBe('');
    attr.sides = [10, 20];
    const angle = Math.PI * attr.angle / 180;
    expect(attr.points).toEqual([0, 0, 10, 0, 20 * Math.cos(angle), 20 * Math.sin(angle)]);
    expect(attr.d).toMatchSnapshot();
    attr.angle = 0;
    expect(attr.points).toEqual([0, 0, 10, 0, 20, 0]);
    expect(attr.d).toMatchSnapshot();
  });
});