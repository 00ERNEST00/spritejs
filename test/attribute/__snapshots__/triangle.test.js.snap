// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`attribute set get get set 1`] = `"M0 0L10 0L10.000000000000002 17.32050807568877Z"`;

exports[`attribute set get get set 2`] = `"M0 0L10 0L20 0Z"`;
